Formula/code-server.rb

```diff
@@ -0,0 +1,78 @@
+class CodeServer < Formula
+  desc "Access VS Code through the browser"
+  homepage "https://coder.com"
+  # We download a built linux release but rebuild the native parts against our version of node/darwin.
```

Comment:
You need to build the entire thing from source.

Reply:
The docs say it's fine to grab things from npm to avoid having to build/bundle/compile javascript which is all this is doing. We're still building all the native modules.

I also looked at other formulas and saw they were doing something similar. See https://github.com/Homebrew/homebrew-core/blob/master/Formula/yarn.rb for example.

Reply:
If it's a node module it should download from the npm registry.

Reply:
We don't have it on there yet.

Reply:
@Homebrew/core opionions?

Reply:
> We don't have it on there yet.

If there is a plan to publish it on npm, then we could wait for it, I think

Reply:
We'd prefer to get in as soon as possible. Could you guys elaborate on your concerns here? The only difference in the build process whether building from npm or from this release is the URL it's fetched from.

Reply:
Wait, I just saw you're part of the project. Gonna decline it for being self-submitted then: https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff

Reply:
It has 30k stars. Does that not cross the threshold for being popular?

Reply:
> Could you guys elaborate on your concerns here?

- That this is using something with `linux-x86_64` sounds like it's downloading binaries. It should be built from the GitHub tag tarball or `npm` instead.
- The install process is pretty messy as-is
- We can consider it if it's self-submitted if some of the mentioned issues are resolved upstream before it is merged
- You may prefer to get it in "as soon as possible" but we want to get it in "as soon as we consider it ready"